src/DaPigGuy/libPiggyEconomy/providers/BedrockEconomyProvider.php

```diff
@@ -4,52 +4,82 @@
 
 namespace DaPigGuy\libPiggyEconomy\providers;
 
-use cooldogedev\BedrockEconomy\api\legacy\ClosureContext;
+use cooldogedev\BedrockEconomy\api\type\ClosureAPI;
 use cooldogedev\BedrockEconomy\BedrockEconomy;
 use cooldogedev\BedrockEconomy\api\BedrockEconomyAPI;
-use cooldogedev\BedrockEconomy\api\version\LegacyBEAPI;
-use cooldogedev\BedrockEconomy\currency\CurrencyManager;
+use cooldogedev\BedrockEconomy\api\util\ClosureWrapper;
+use cooldogedev\BedrockEconomy\currency\Currency;
 use pocketmine\player\Player;
 use pocketmine\Server;
 
 class BedrockEconomyProvider extends EconomyProvider
 {
-    private LegacyBEAPI $api;
-    private CurrencyManager $currency;
+    private ClosureAPI $api;
+    private Currency $currency;
 
     public static function checkDependencies(): bool
     {
-        return Server::getInstance()->getPluginManager()->getPlugin("BedrockEconomy") !== null;
+        $plugin = Server::getInstance()->getPluginManager()->getPlugin("BedrockEconomy");
+        return $plugin !== null && version_compare($plugin->getDescription()->getVersion(), '4.0', '>=');
     }
 
     public function __construct()
     {
-        $this->api = BedrockEconomyAPI::legacy();
-        $this->currency = BedrockEconomy::getInstance()->getCurrencyManager();
+        $this->api = BedrockEconomyAPI::CLOSURE();
+        $this->currency = BedrockEconomy::getInstance()->getCurrency();
     }
 
     public function getMonetaryUnit(): string
     {
-        return $this->currency->getSymbol();
+        return $this->currency->symbol;
     }
 
     public function getMoney(Player $player, callable $callback): void
     {
-        $this->api->getPlayerBalance($player->getName(), ClosureContext::create(fn(?int $balance) => $callback($balance ?? $this->currency->getDefaultBalance())));
+        $this->api->get(
```

Comment:
`ClosureAPI.get` is [marked internal](https://github.com/cooldogepm/BedrockEconomy/blob/pm5/src/cooldogedev/BedrockEconomy/api/type/ClosureAPI.php#L74)